cmd/soroban-cli/src/config/sc_address.rs

```diff
@@ -0,0 +1,57 @@
+use std::str::FromStr;
+
+use crate::xdr;
+
+use super::{address, locator, ContractAddress};
+
+/// `ScAddress` can be either a resolved `xdr::ScAddress` or an alias of a `Contract` or `MuxedAccount`.
+#[derive(Clone, Debug)]
+pub enum ScAddress {
```

Comment:
It's pretty confusing to me that we have ScAddress that overlaps with name of the actual ScAddress. It's also confusing that the `config::Address` wrapped a muxed and alias.

Naming here is going to be hard.

I suggest we rename both:

- `config::Address` -> `config::MuxedAccountOrAlias`
- `config::ScAddress` -> `config::ScAddressOrAlias`

Reply:
Or, actually:
- `config::Address` -> `config::UnresolvedMuxedAccount`
- `config::ScAddress` -> `config::UnresolvedScAddress`